### DISCUSSION TITLE
编译后version信息缺失

### DISCUSSION THREAD

**Discussion by lsCoding666** [AlistGo/alist#888] [Q&A]
按照官网和build.sh的脚本，编译结束后运行发现版本号出问题，如图所示
![image](https://user-images.githubusercontent.com/73888354/161372724-b2f2d842-3727-48bf-9cce-58e55f904a3e.png)
这个是正常的
![image](https://user-images.githubusercontent.com/73888354/161372747-d9a5f1ca-295f-418c-9acf-faad2cbfca0d.png)
我觉得好像是webTag出了问题，编译后手动在data/config.json指定版本后正常
![image](https://user-images.githubusercontent.com/73888354/161372790-10728459-39bc-4108-891d-96d034bd6412.png)
![image](https://user-images.githubusercontent.com/73888354/161372857-a5972729-2f50-40ae-860a-7320658c900a.png)

**Comment by xhofe** [comment]
一般来说，自己编译前后端，前端是无法使用cdn的。。你这是只编译了后端？前端还是下载的release里的？

**Reply by xhofe** [reply]
如果你确实要这么做的话 可以参照https://github.com/Xhofe/alist/blob/342729179ddeceaff4bbc5c1832ee41c6e5331ff/build.sh#L52 加入web版本信息。

**Reply by lsCoding666** [reply]
编译操作执行过程 ，先编译前端 ：
clone后yarn & yarn run build.
然后将dist文件复制到后端的public文件夹内
执行下面的命令：
```
appName="alist"
builtAt="$(date +'%F %T %z')"
goVersion=$(go version | sed 's/go version //')
gitAuthor=$(git show -s --format='format:%aN <%ae>' HEAD)
gitCommit=$(git log --pretty=format:"%h" -1)
gitTag=$(git describe --long --tags --dirty --always)
ldflags="\
-w -s \
-X 'github.com/Xhofe/alist/conf.BuiltAt=$builtAt' \
-X 'github.com/Xhofe/alist/conf.GoVersion=$goVersion' \
-X 'github.com/Xhofe/alist/conf.GitAuthor=$gitAuthor' \
-X 'github.com/Xhofe/alist/conf.GitCommit=$gitCommit' \
-X 'github.com/Xhofe/alist/conf.GitTag=$gitTag' \
"
go build -ldflags="$ldflags" alist.go
```
编译生成二进制文件，执行该二进制文件出现以上错误。
另外，运行
```webTag=$(wget -qO- -t1 -T2 "https://api.github.com/repos/alist-org/alist-web/releases/latest" | grep "tag_name" | head -n 1 | awk -F ":" '{print $2}' | sed 's/\"//g;s/,//g;s/ //g') ```
无返回值
![image](https://user-images.githubusercontent.com/73888354/161377287-59b39704-39b9-49a3-becb-0e8e5610f8a6.png)

**Reply by lsCoding666** [reply]
后来改了前端代码，把  assets 路径换成本地的了，因为改了代码之后就用不了cdn的了，就用本地正常了。我觉得提这个讨论让人感觉懵的是一般直接执行项目根目录的build.sh或者是参照官方文档的步骤是能正常运行的，但是alist参照官方文档来貌似不行，还得改其他地方。
没改代码前直接按照官方文档来也跑不起来，感觉就有些异常，所以来提单
https://alist-doc.nn.ci/docs/install/source-code

**Reply by xhofe** [reply]
本身自己编译就用不了 cdn 使用需要发布到npm。

**Reply by xhofe** [reply]
按照文档编译是可以用的 只是assets只能用本地而已。
